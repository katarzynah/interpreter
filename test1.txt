program f;
var i, j, k: int;
begin
i := 12;
j := i + 5;
k := 6 * j;
end.