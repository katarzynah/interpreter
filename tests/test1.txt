program f;
var i, j, k, l: int;
s: string;
b : bool;

procedure hej(x : int);
var y: string;
begin
print(y);
end;

begin
i := 0;
i := i + 1;
print(i);
j := 0;
k := 0;
b := k >= i;
l := 0;
s := "hej";
print(s);
print(j + k);
for i := 0 to j + k do
begin
    print (i);
    l := l + 1;
end;
print(b);
print(l);

print(i);
print(k);
while i > k do
    i := i - 1;
print(i);

if true then print("hej") endif;
if i = k then print("equal") endif;
end.