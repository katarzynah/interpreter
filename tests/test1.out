[(fromList [(Ident "b",undefined),(Ident "i",undefined),(Ident "j",undefined),(Ident "k",undefined),(Ident "l",undefined),(Ident "s",undefined)],fromList [(Ident "hej",ProcDecProc (ProcHead (Ident "hej") (Args (ArgList (ArgLabel (IdLEnd (Ident "x")) TypeSpecInt) ArgListEmpty))) (VarDecFull (VarDecListEnd (VarDecLabel (IdLEnd (Ident "y")) TypeSpecString))) (CompStmnt (StmntList (SPrint (PrintStmnt (ExpSimple (SimpleExpTerm (TermFactor (FactorExpression (ExpSimple (SimpleExpTerm (TermFactor (FactorIdent (Ident "y"))))))))))) StmntListEmpty)))])]
1
"hej"
0
0
False
1
1
0
0
"hej"
"equal"
